tests/testthat/test-parse.R

```diff
@@ -25,13 +25,67 @@ test_that("coerce_datetime fills the void", {
   expect_error(coerce_datetime(NA_complex_, NA_datetime_, name = "complexity"), class = "vctrs_error_incompatible_type")
 })
 
-test_that("make_timestamp works with POSIXct", {
-  outcome <- "2020-01-01T01:02:03Z"
+test_that("parse_connect_rfc3339 parses timestamps we expect from Connect", {
+  original_tz <- Sys.getenv("TZ")
+  withr::defer(Sys.setenv(TZ = original_tz))
+
+  xs <- c(
```

Comment:
If my suggestion above is correct, then you'd need to revise these tests too, IIUC.

Reply:
I was gonna say, yes, the tests will need to be reworked, but that's doable.

Reply:
As I said above, I'm ok with leaving the more flexible code in place that you have, but let's do rewrite the tests so that they reflect the behavior that Connect has, i.e. we don't have heterogeneous timestamp formatting in a single vector. That's not a behavior we need to support, so let's not write tests that codify that as behavior that is important, so that whenever we refactor this in the future, we don't imagine that to be a feature requirement.

Reply:
Tests now include single-time-zone vectors with multiple and single elements, offsets and zero offsets.